incubator/hnc/test/e2e/quickstart_test.go

```diff
@@ -137,33 +139,35 @@ spec:
   - from:
     - podSelector: {}`
 
-		filename := WriteTempFile(policy)
-		defer RemoveFile(filename)
-		MustRun("kubectl apply -f", filename)
-		// ensure this policy can be propagated to its descendants
+		MustApplyYAML(policy)
+		// Enable propagation for netpols and wait for it to get propagated at least to service-1
 		MustRun("kubectl hns config set-resource networkpolicies --group networking.k8s.io --mode Propagate --force")
-		expected := "deny-from-other-namespaces"
-		RunShouldContain(expected, defTimeout, "kubectl get netpol -n", nsOrg)
-		RunShouldContain(expected, defTimeout, "kubectl get netpol -n", nsTeamA)
-		RunShouldContain(expected, defTimeout, "kubectl get netpol -n", nsTeamB)
-		RunShouldContain(expected, defTimeout, "kubectl get netpol -n", nsService1)
-		RunShouldContain(expected, defTimeout, "kubectl get netpol -n", nsService2)
+		RunShouldContain("deny-from-other-namespaces", defTimeout, "kubectl get netpol -n", nsService1)
 
 		// Now we’ll see that we can no longer access service-2 from the client in service-1. If we can,
 		// that probably means that network policies aren't enabled on this cluster (e.g. Kind, GKE by
 		// default) and we should skip the rest of this test.
-		netpolTestStdout := ""
-		Eventually(func() error {
-			stdout, err := RunCommand("kubectl run client -n", nsService1, clientArgs, cmdln)
-			netpolTestStdout = stdout
-			return err
-		}).Should(Succeed())
-		if !strings.Contains(netpolTestStdout, "wget: download timed out") {
+		//
+		// The standard matching functions won't work here because we're looking for a particular error
+		// string, but we don't want to fail if we've found it. So use the default timeout (2s) by
+		// trying up to three times with a 1s gap in between.
+		netpolWorks := false
+		for i:=0; !netpolWorks && i<3; i++ {
+			// This command will return a non-nil error if it works correctly
+			stdout, _ := RunCommand(clientCmd, nsService1, alpineArgs, wgetArgs)
+			if strings.Contains(stdout, "wget: download timed out") {
```

Comment:
Question: If the netpol is not enabled in the cluster it will just get "Welcome to nginx!", right?